src/OracleService/Protocols/OracleNeoFSProtocol.cs

```diff
@@ -0,0 +1,139 @@
+using Google.Protobuf;
+using Neo.Cryptography.ECC;
+using Neo.Network.P2P.Payloads;
+using Neo.Wallets;
+using NeoFS.API.v2.Client;
+using NeoFS.API.v2.Client.ObjectParams;
+using NeoFS.API.v2.Cryptography;
+using NeoFS.API.v2.Refs;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Object = NeoFS.API.v2.Object.Object;
+using Range = NeoFS.API.v2.Object.Range;
+
+namespace Neo.Plugins
+{
+    class OracleNeoFSProtocol : IOracleProtocol
+    {
+        private readonly byte[] privateKey;
+        private const string URIScheme = "neofs";
+        private const string RangeCmd = "range";
+        private const string HeaderCmd = "header";
+        private const string HashCmd = "hash";
+
+        public void Configure()
+        {
+        }
+
+        public OracleNeoFSProtocol(Wallet wallet, ECPoint[] oracles)
+        {
+            privateKey = oracles.Select(p => wallet.GetAccount(p)).Where(p => p is not null && p.HasKey && !p.Lock).FirstOrDefault()?.GetKey().PrivateKey;
+        }
+
+        public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, CancellationToken cancellation)
+        {
+            Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"Request: {uri.AbsoluteUri}");
+
+            if (!Settings.Default.AllowPrivateHost)
+            {
+                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host);
+                if (entry.IsInternal())
+                    return (OracleResponseCode.Forbidden, null);
+            }
+            try
+            {
+                byte[] res = Get(privateKey, uri, Settings.Default.NeoFS.EndPoint, cancellation);
+                Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"NeoFS result: {res.ToHexString()}");
+                return (OracleResponseCode.Success, Convert.ToBase64String(res));
+            }
+            catch (Exception e)
+            {
+                Utility.Log(nameof(OracleNeoFSProtocol), LogLevel.Debug, $"NeoFS result: error,{e}");
+                return (OracleResponseCode.Error, null);
+            }
+        }
+
+        private byte[] Get(byte[] privateKey, Uri uri, string host, CancellationToken cancellation)
+        {
+            string[] ps = uri.PathAndQuery.TrimStart('/').Split("/");
+            if (ps.Length == 0) throw new Exception("object ID is missing from URI");
+            ContainerID containerID = ContainerID.FromBase58String(uri.OriginalString.Substring((URIScheme + "://").Length, uri.Host.Length));
```

Comment:
There is no host in neofs uri. The format of neofs uri is:

```
neofs:{containerID}/{objectID}
```

Reply:
OK.I will fix it.  @roman-khimov      I think it also need to fix here.
https://github.com/nspcc-dev/neo-go/pull/1484

Reply:
We're just misusing standard `net/url` package a little for interface consistency across different oracle types, maybe we'll change the interface a little or leave it as is (it works this way).

Reply:
We need to fix it.